jbehave-support-core/src/main/java/org/jbehavesupport/core/report/extension/JmsXmlReporterExtension.java

```diff
@@ -0,0 +1,83 @@
+package org.jbehavesupport.core.report.extension;
+
+import lombok.extern.slf4j.Slf4j;
+import org.jbehavesupport.core.report.ReportContext;
+
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.TextMessage;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Slf4j
+public class JmsXmlReporterExtension extends AbstractXmlReporterExtension {
+
+    private static final String JMS_XML_REPORTER_EXTENSION = "jms";
+    private static final String MESSAGE_TAG = "message";
+    private static final String CID_TAG = "cid";
+    private static final String ANSWERS_TAG = "answers";
+    private static final String ROW_TAG = "row";
+    private static final String COLUMN_TAG = "column";
+    private static final String STRING_TAG = "toString";
+
+    private static final String[] headers = {"JMSMessageID", "JMSDestination", "JMSDeliveryMode", "JMSExpiration", "JMSPriority", "JMSTimestamp", "JMSReplyTo", "JMSType", "JMSRedelivered"};
+
+    private final List<Message> messages = new ArrayList();
+
+    @Override
+    public String getName() {
+        return JMS_XML_REPORTER_EXTENSION;
+    }
+
+    public void report(Message message) {
+        messages.add(message);
+    }
+
+    @Override
+    public void print(Writer writer, ReportContext reportContext) {
+        messages.forEach(message -> printJmsMessage(writer, message));
+        messages.clear();
+    }
+
+    public void printJmsMessage(Writer writer, Message message) {
+        try {
+            printBegin(writer, MESSAGE_TAG);
+            printBegin(writer, CID_TAG);
+            printCData(writer, message.getJMSCorrelationID());
+            printEnd(writer, CID_TAG);
+
+            printBegin(writer, ANSWERS_TAG);
+            Arrays.stream(headers).forEach(header -> {
+                printBegin(writer, ROW_TAG);
+                printBegin(writer, COLUMN_TAG);
+                printCData(writer, header);
+                printEnd(writer, COLUMN_TAG);
+
+                try {
+                    printBegin(writer, COLUMN_TAG);
+                    printCData(writer, message.getStringProperty(header));
+                    printEnd(writer, COLUMN_TAG);
+                } catch (JMSException e) {
+                    log.error("Unable to get the property: {}", header);
```

Comment:
Maybe we could print that inside the the tag? not sure, just something to think about if it makes sense or not in this case (+ maybe log the exception?)